### PR TITLE
Setup the initial file system

### LINKED ISSUE #45: Create Initial File System
### Steps
- [x] Identify all necessary directories and files for the project
- [x] Define a clear and logical file system structure for code, configurations, scripts, and documentation
- [x] Create the directories and files based on the defined structure
- [x] Add gitignore
- [x] Add readme

### User Story
As a developer, I need a well-organized file system structure to ensure that all project files are systematically arranged for easy navigation, maintainability, and collaboration.

### PR DESCRIPTION
**Pull request by BerkayDur**
Setup the initial file systems to create a separate directory for each task. This includes adding: `README.md` to each directory and a global `.gitignore`. Closes #45

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+*.parquet
+data
+
+# venv
+.venv
+
+# env
+.env
+
+# terraform
```
Comment (danfh00) [inline .gitignore:12]: Excellent use of * for varying filenames

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# data files/directories
```
Comment (A-Faris) [inline .gitignore:1]: Good comment, makes it clearer to understand

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+# venv
+.venv
+
+# env
+.env
+
+# terraform
+.terraform*
+*.tfvars
+*.tfstate*
```
Comment (A-Faris) [inline .gitignore:15]: Why is there * at the end of this?

**Review by danfh00 — APPROVED** [review]
Excellent Work Berkay💪

**Review by A-Faris — COMMENTED** [review]
File structure looks good

File: .gitignore (hunk 3 above)
Comment (BerkayDur) [reply .gitignore:15]: Terraform creates a `.terraform.lock.hcl` which contains sensitive data. So, this is a universal catch-up on files of this sort.

Comment (A-Faris) [reply .gitignore:15]: interesting

**Pull request merged**